pkg/app/piped/executor/lambda/lambda.go

```diff
@@ -82,16 +84,65 @@ func decideRevisionName(in *executor.Input, fm provider.FunctionManifest, commit
 	return
 }
 
-func apply(ctx context.Context, in *executor.Input, cloudProviderName string, cloudProviderCfg *config.CloudProviderLambdaConfig, fm provider.FunctionManifest) bool {
+func sync(ctx context.Context, in *executor.Input, cloudProviderName string, cloudProviderCfg *config.CloudProviderLambdaConfig, fm provider.FunctionManifest) bool {
 	in.LogPersister.Infof("Start applying the lambda function manifest")
 	client, err := provider.DefaultRegistry().Client(cloudProviderName, cloudProviderCfg, in.Logger)
 	if err != nil {
 		in.LogPersister.Errorf("Unable to create Lambda client for the provider %s: %v", cloudProviderName, err)
 		return false
 	}
 
-	if err := client.Apply(ctx, fm); err != nil {
-		in.LogPersister.Errorf("Failed to apply the lambda function manifest (%v)", err)
+	ok, err := client.AvailableFunctionName(ctx, fm.Spec.Name)
+	if err != nil {
+		in.LogPersister.Errorf("Unable to validate function name %s: %v", fm.Spec.Name, err)
+		return false
+	}
+	if ok {
+		if err := client.CreateFunction(ctx, fm); err != nil {
+			in.LogPersister.Errorf("Failed to create lambda function %s: %v", fm.Spec.Name, err)
+			return false
+		}
+	} else {
+		if err := client.UpdateFunction(ctx, fm); err != nil {
+			in.LogPersister.Errorf("Failed to update lambda function %s: %v", fm.Spec.Name, err)
+			return false
+		}
+	}
+
+	// Wait before ready to commit change.
+	in.LogPersister.Info("Waiting to update lambda function in progress...")
+	time.Sleep(3 * time.Minute)
+
+	// Commit version for applied Lambda function.
+	version, err := client.PublishFunction(ctx, fm)
```

Comment:
I'm concerned that when users clicked the SYNC button to sync the application without any changes.
In that case, no changes were added on AWS Lambda side and it is ok to publish the function?
I think we may receive a "NoChange" error or something like that.

Reply:
> AWS Lambda doesn't publish a version if the function's configuration and code haven't changed since the last version. Use UpdateFunctionCode or UpdateFunctionConfiguration to update the function before publishing a version.

Yes, you guess right, due to the docs of [PublishVersion](https://docs.aws.amazon.com/sdk-for-go/api/service/lambda/#Lambda.PublishVersion), we have to make some changes on source/configuration of the function to be able to publish a new version of it. So the error returned here for that case is predictable. 👍

Reply:
Added a note about this lack on the AWS documentation.